shared/js/contacts/contacts_matcher.js

```diff
@@ -1,710 +1,7 @@
-      }
-      else {
-        finalResult[aResult.id].matchings.name =
-          matchingObj.matchings.name;
-      }
-    });
-
-    if (Object.keys(finalResult).length > 0) {
-      notifyMatch(callbacks, finalResult);
-    }
-    else {
-      notifyMismatch(callbacks);
-    }
-  }
-
-  function getCompleteName(contact) {
-    var givenName = Array.isArray(contact.givenName) ?
-                    contact.givenName[0] : '';
-
-    var familyName = Array.isArray(contact.familyName) ?
-                    contact.familyName[0] : '';
-
-    var completeName = givenName && familyName ?
-                       givenName + ' ' + familyName :
-                       givenName || familyName;
-
-    return completeName;
-  }
-
-  function isEmpty(collection) {
-    return ((!Array.isArray(collection) || !collection[0]) || (collection[0] &&
-                          collection[0].value && !collection[0].value.trim()));
-  }
-
-  function isEmptyStr(collection) {
-    return (!Array.isArray(collection) ||
-            (typeof collection[0] !== 'string') || !(collection[0].trim()));
-  }
-
-  function hasName(aContact) {
-    return (!isEmptyStr(aContact.givenName) ||
-            !isEmptyStr(aContact.familyName));
-  }
-
-  function reconcileResults(incomingContact, nameMatches, phoneMailMatches,
-                            callbacks) {
-    var finalMatchings = {};
-
-    // Name matches drive all the process
-    Object.keys(nameMatches).forEach(function(aNameMatching) {
-      var matchingContact = nameMatches[aNameMatching].matchingContact;
-
-      var isPhoneMatching = Array.isArray(phoneMailMatches[aNameMatching].
-                                          matchings.tel);
-      var isMailMatching = Array.isArray(phoneMailMatches[aNameMatching].
-                                          matchings.email);
-
-      // Three cases under which a matching is considered
-      if (isPhoneMatching && isMailMatching) {
-        finalMatchings[aNameMatching] = phoneMailMatches[aNameMatching];
-      }
-      else if (isPhoneMatching && (isEmpty(incomingContact.email) ||
-              isEmpty(matchingContact.email))) {
-        finalMatchings[aNameMatching] = phoneMailMatches[aNameMatching];
-      }
-      else if (isMailMatching && (isEmpty(incomingContact.tel) ||
-                                  isEmpty(matchingContact.tel))) {
-        finalMatchings[aNameMatching] = phoneMailMatches[aNameMatching];
-      }
-    });
-
-    if (Object.keys(finalMatchings).length > 0) {
-      notifyMatch(callbacks, finalMatchings);
-    }
-    else {
-      notifyMismatch(callbacks);
-    }
-  }
-  
-  // Exported method
-  this.match = doMatch;
-  
-  Object.defineProperty(this, 'dataProvider', {
-    set: function(theProvider) {
-      dataProvider = theProvider;
-    }
-  });
-};
-
-contacts.Matcher = new contacts.MatcherObj(navigator.mozContacts);
+(function(exports) {
+  exports.Matcher = new MatcherObj(navigator.mozContacts);
```

Comment:
Also if it's necessary to expose the MatcherObject, that can be done here, a file can export more than one class.